File: test-image.py
import base64
import httpx
import os
from letta_client import Letta

token = os.getenv('LETTA_API_KEY')
if not token:
    print("Error: LETTA_API_KEY environment variable not set")
    exit(1)

client = Letta(token=token)

image_url = "https://upload.wikimedia.org/wikipedia/commons/a/a7/Camponotus_flavomarginatus_ant.jpg"
image_data = base64.standard_b64encode(httpx.get(image_url).content).decode("utf-8")

response = client.agents.messages.create(
    agent_id="agent-bb780791-961a-4fa3-95ba-b681b6d508e6",
    messages=[
        {
            "role": "user",
            "content": [
                {
                    "type": "image",
                    "source": {
                        "type": "base64",
                        "media_type": "image/jpeg",
                        "data": image_data,
                    },
                },
                {
                    "type": "text",
                    "text": "Describe this image."
                }
            ],
        }
    ],
)

print("Response:")
print(response)

File: test-image.ts
import { LettaClient } from '@letta-ai/letta-client';

async function testImage() {
    const token = process.env.LETTA_API_KEY;
    if (!token) {
        console.error('Error: LETTA_API_KEY environment variable not set');
        process.exit(1);
    }

    const client = new LettaClient({ token });

    const imageUrl = "https://upload.wikimedia.org/wikipedia/commons/a/a7/Camponotus_flavomarginatus_ant.jpg";
    const imageResponse = await fetch(imageUrl);
    const imageBuffer = await imageResponse.arrayBuffer();
    const imageData = Buffer.from(imageBuffer).toString('base64');

    const response = await client.agents.messages.create(
        "agent-bb780791-961a-4fa3-95ba-b681b6d508e6", {
            messages: [
                {
                    role: "user",
                    content: [
                        {
                            type: "text",
                            text: "Describe this image."
                        },
                        {
                            type: "image",
                            source: {
                                type: "base64",
                                mediaType: "image/jpeg",
                                data: imageData,
                            },
                        }
                    ],
                }
            ],
        }
    );

    console.log('Response:', response);
}

testImage().catch(console.error);
